engine/consensus/compliance/engine.go

```diff
@@ -51,6 +49,7 @@ func NewEngine(
 	log zerolog.Logger,
 	me module.Local,
 	core *Core,
+	actor *events.FinalizationActor,
```

Comment:
I am not sure about passing actor as argument, suddenly we are injecting dependency instead of using pub-sub. 
How about embedding `actor` in structure and subscribing externally on `FinalizationActor.OnBlockFinalized`. This way we will get benefits from using `FinalizationActor` but still preserve a pub-sub style subscription to avoid hard dependency.

Reply:
Changed the construction pattern in https://github.com/onflow/flow-go/pull/4187/commits/b8d38e6bec80ced298aeecceff262df23ba350ee